detox/src/devices/drivers/android/AndroidDriver.js

```diff
@@ -289,27 +282,31 @@ class AndroidDriver extends DeviceDriverBase {
     return this.invocationManager.execute(DetoxApi.launchMainActivity());
   }
 
-  async _waitForProcess(deviceId, bundleId) {
+  async _waitForProcess(adbName, bundleId) {
     let pid = NaN;
     try {
-      const queryPid = () => this._queryPID(deviceId, bundleId);
+      const queryPid = () => this._queryPID(adbName, bundleId);
       const retryQueryPid = () => retry({ backoff: 'none', retries: 4 }, queryPid);
       const retryQueryPidMultiple = () => retry({ backoff: 'linear' }, retryQueryPid);
       pid = await retryQueryPidMultiple();
     } catch (e) {
-      log.warn(await this.adb.shell(deviceId, 'ps'));
+      log.warn(await this.adb.shell(adbName, 'ps'));
       throw e;
     }
     return pid;
   }
 
-  async _queryPID(deviceId, bundleId) {
-    const pid = await this.adb.pidof(deviceId, bundleId);
+  async _queryPID(adbName, bundleId) {
+    const pid = await this.adb.pidof(adbName, bundleId);
     if (!pid) {
       throw new Error('PID still not available');
     }
     return pid;
   }
+
+  _getAdbName(deviceId) {
+    return _.isObjectLike(deviceId) ? deviceId.adbName : deviceId;
+  }
```

Comment:
As explained in the description, it is no longer safe to assume that `deviceId` and `adbName` are interchangeable.

Reply:
We no longer can call it `deviceId`, after this transformation. I suggest renaming it to `deviceHandle`.